torchvision/models/detection/ssd.py

```diff
@@ -164,27 +164,39 @@ class SSD(nn.Module):
         'proposal_matcher': det_utils.Matcher,
     }
 
-    def __init__(self, backbone: nn.Module, anchor_generator: DefaultBoxGenerator,
-                 size: Tuple[int, int], num_classes: int,
-                 image_mean: Optional[List[float]] = None, image_std: Optional[List[float]] = None,
-                 head: Optional[nn.Module] = None,
-                 score_thresh: float = 0.01,
-                 nms_thresh: float = 0.45,
-                 detections_per_img: int = 200,
-                 iou_thresh: float = 0.5,
-                 topk_candidates: int = 400,
-                 positive_fraction: float = 0.25):
-        super().__init__()
+    def __init__(
+        self,
+        backbone: nn.Module,
+        anchor_generator: DefaultBoxGenerator,
+        size: Tuple[int, int],
+        num_classes: int,
+        image_mean: Optional[List[float]] = None,
+        image_std: Optional[List[float]] = None,
+        head: Optional[nn.Module] = None,
+        score_thresh: float = 0.01,
+        nms_thresh: float = 0.45,
+        detections_per_img: int = 200,
+        iou_thresh: float = 0.5,
+        topk_candidates: int = 400,
+        positive_fraction: float = 0.25
+    ) -> None:
 
+        super().__init__()
         self.backbone = backbone
-
         self.anchor_generator = anchor_generator
-
         self.box_coder = det_utils.BoxCoder(weights=(10., 10., 5., 5.))
 
         if head is None:
             if hasattr(backbone, 'out_channels'):
-                out_channels = backbone.out_channels
+                # Other models use int as out_channel, we need compatibility.
```

Comment:
Some patchy workaround for mypy.
Since in other models we use `out_channels` from backbone as int, and in this case we need `List[int]`.
We can handle the single int case by simply converting to list. Bit consistency for the user.

Reply:
1. If `out_channels` can be `int` or `List[int]`, shouldn't we also cast it as `Union[int, List[int]]`?
2. From the old code I get that `out_channels` is expected to be a `List[int]` so adding the possibility of it being `int` is a runtime change. Let's just annotate in this PR and only change stuff in follow-up PRs.